### PR TITLE
feat(17482): Upload MCDA button added to toolbar. Upload MCDA flow added

### PR DESCRIPTION
**Pull request by albaranau**
- Upload MCDA button added to toolbar https://kontur.fibery.io/Tasks/Task/Add-Upload-MCDA-button-to-toolbar-17482

- Upload MCDA flow added https://kontur.fibery.io/Tasks/Task/Add-Upload-MCDA-flow-17193

## Summary by CodeRabbit

- **New Features**
	- Introduced Multi-Criteria Decision Analysis (MCDA) support in the toolbar.
	- Added functionality for uploading and validating MCDA JSON files.
	- Enhanced MCDA feature with initialization controls and state management for better user interaction.
- **Refactor**
	- Streamlined constants usage for MCDA feature to ensure consistency and maintainability.

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -11476,7 +11491,7 @@ packages:
     engines: {node: '>=12'}
     dev: false
 
-  /webpack-dev-middleware@4.3.0(webpack@5.90.0):
+  /webpack-dev-middleware@4.3.0(webpack@5.90.1):
```
Comment (amurKontur) [inline pnpm-lock.yaml:11494]: can this updates affect some areas of the app and if so, then what areas?

File: src/features/mcda/mcdaConfig.tsx [hunk 2]
```diff
@@ -75,3 +75,14 @@ function createMCDALayersFromBivariateAxises(axises: Axis[]): MCDALayer[] {
     return acc;
   }, []);
 }
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function isMCDAConfig(json: any): json is MCDAConfig {
+  // TODO: full check using ajv
+  return (
+    json.id &&
```
Comment (amurKontur) [inline src/features/mcda/mcdaConfig.tsx:83]: why not _json?.id_    ?

File: src/features/mcda/openMcdaFile.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import { currentNotificationAtom } from '~core/shared_state';
+import { isMCDAConfig } from './mcdaConfig';
+import type { MCDAConfig } from '~core/logical_layers/renderers/stylesConfigs/mcda/types';
+
+const input = (() => {
+  const input = document.createElement('input');
+  input.type = 'file';
+  input.multiple = false;
+  return input;
+})();
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply src/features/mcda/openMcdaFile.ts:11]: it is an arrow function used, need to mark it as irrelevant. What ecmascript version does coderabbitai follow?

**Review by akolesen — APPROVED** [review]

File: src/features/mcda/openMcdaFile.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import { i18n } from '~core/localization';
+import { currentNotificationAtom } from '~core/shared_state';
+import { isMCDAConfig } from './mcdaConfig';
+import type { MCDAConfig } from '~core/logical_layers/renderers/stylesConfigs/mcda/types';
+
+const input = (() => {
+  const input = document.createElement('input');
+  input.type = 'file';
```
Comment (amurKontur) [inline src/features/mcda/openMcdaFile.ts:8]: maybe add max limit file size ?

File: src/features/mcda/openMcdaFile.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  input.multiple = false;
+  return input;
+})();
+
+function readMcdaJSONFile(file): Promise<MCDAConfig> {
+  return new Promise((res, rej) => {
+    const reader = new FileReader();
+    reader.onload = function (event) {
+      if (event.target === null) return;
+      const s = event.target.result?.toString();
```
Comment (amurKontur) [inline src/features/mcda/openMcdaFile.ts:18]: what is s? maybe another name of const should be used?

File: src/features/mcda/openMcdaFile.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+      }
+    };
+    reader.onerror = (error) => rej(error);
+    reader.readAsText(file);
+  });
+}
+
+export function askMcdaJSONFile(onSuccess: (mcdaConfig: MCDAConfig) => void) {
+  async function onchange() {
+    if ('files' in input && input.files !== null) {
```
Comment (amurKontur) [inline src/features/mcda/openMcdaFile.ts:38]: if (input.files ?? false) 

in case we use at least ecmascript 2020

**Review by amurKontur — APPROVED** [review]

**Pull request merged**